### PR TITLE
Task3

### PR DESCRIPTION
**Pull request by LesterOper**

### CODE REVIEW

File: src/main/java/com/es/phoneshop/Cart/Cart.java [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+import java.util.List;
+
+public class Cart {
+    private List<CartItem> cart;
+    
+    public Cart(){
+        cart = new ArrayList<>();
+    }
+    
+    public List<CartItem> getList(){
```
Comment (SergeiStarikov) [inline src/main/java/com/es/phoneshop/Cart/Cart.java:13]: Here and in other places - coding style guides:

getList(){ --> getList() {
if(session.getAttribute(VIEWED_ITEMS)==null){ --> if (session.getAttribute(VIEWED_ITEMS) == null) {

Style guides: https://google.github.io/styleguide/javaguide.html

Let IDEA format the file for you: https://www.jetbrains.com/help/idea/reformat-file-dialog.html

Ctrl + Shift + Alt + L or appropriate option in settings

File: src/main/java/com/es/phoneshop/Cart/HttpSessionCartServiceService.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            cart = (Cart)session.getAttribute(CART_ATTRIBUTE);
+        }
+        return cart;
+    }
+
+    @Override
+    public void addProduct(Cart cartList,Product product, int quantity) {
+        if(cartList.getList().stream().anyMatch(prod-> prod.getProduct().getId().equals(product.getId()))){
+            cartList.getList().stream().
+                    filter(prod-> prod.getProduct().getId().equals(product.getId())).
```
Comment (SergeiStarikov) [inline src/main/java/com/es/phoneshop/Cart/HttpSessionCartServiceService.java]: Streams formatting:

 cartList.getList().stream()
                    .filter()
                    .map()

File: src/main/webapp/WEB-INF/pages/productDescription.jsp [hunk 3]
```diff
@@ -15,7 +13,7 @@
   </p>
   <p>
     <c:if test="${error!=null}">
-        <font color ="red"> <c:out value="You enter smth wrong"/></font>
+        <span style="color: red; "><c:out value="You enter smth wrong"/></span>
```
Comment (SergeiStarikov) [inline src/main/webapp/WEB-INF/pages/productDescription.jsp]: Here and in other places - HTML formatting:
<span>
    <...>
</span>

File: src/main/webapp/WEB-INF/pages/productDescription.jsp [hunk 4]
```diff
@@ -38,24 +36,45 @@
...
-        </c:if>
-        <c:if test="${error==null}">
+        </form>
         <p>
-            <font color="green">
-            <c:out value ="${param.message}"/>
-            </font>
+            <c:choose>
+                <c:when test="${error!=null}">
+                    <span style="color: red; "><c:out value="${error}"/></span>
```
Comment (SergeiStarikov) [inline src/main/webapp/WEB-INF/pages/productDescription.jsp]: Please move inline styles to .css file

File: src/main/webapp/WEB-INF/pages/productDescription.jsp [hunk 5]
```diff
@@ -38,24 +36,45 @@
...
+                        <a href="products/${view.id}">${view.description}</a>
+                    </p>
+                    <p class="price">
+                        <fmt:formatNumber value="${view.price}" type="currency" currencySymbol="${view.currency.symbol}"/>
+                    </p>
+                </td>
+            </table>
+            </p>
+            </c:forEach>
+        </c:if>
```
Comment (SergeiStarikov) [inline src/main/webapp/WEB-INF/pages/productDescription.jsp]: Formatting again:

Compare what it takes to read this:

```
<c:set var="list" value="${sessionScope['cartList'].list}"/>
        <c:if test="${list!=null}">
            <c:forEach items="${list}" var="cart" >
                <p><c:out value="${cart}"/></p>
            </c:forEach>
        </c:if>
    <c:set var="list" value="${sessionScope['viewed'].list}"/>
        <c:if test="${list!=null}">
            <c:forEach items="${list}" var="view">
            <p>
                ...
            </p>
            </c:forEach>
        </c:if>
```

and this:

```
<c:set var="list" value="${sessionScope['cartList'].list}"/>
<c:set var="list" value="${sessionScope['viewed'].list}"/>

<c:if test="${list!=null}">
    <c:forEach items="${list}" var="cart">
         <p>
             <c:out value="${cart}"/>
         </p>
    </c:forEach>
</c:if>

<c:if test="${list!=null}">
    <c:forEach items="${list}" var="view">
        <p>
            ...        
        </p>
    </c:forEach>
<c:if>
```

File: src/main/webapp/WEB-INF/pages/productDescription.jsp [hunk 6]
```diff
@@ -38,24 +36,45 @@
...
         <c:if test="${list!=null}">
             <c:forEach items="${list}" var="cart" >
                 <p><c:out value="${cart}"/></p>
             </c:forEach>
-        </c:if>      
+        </c:if>
+    <c:set var="list" value="${sessionScope['viewed'].list}"/>
+        <c:if test="${list!=null}">
+            <c:forEach items="${list}" var="view">
+            <p>
```
Comment (SergeiStarikov) [inline src/main/webapp/WEB-INF/pages/productDescription.jsp:67]:
```
<p>
            <table>
                <td>
                   <p>
```

Do we need this? table inside p, p inside td. No <tr> tag

Please check proper table layout: https://www.w3schools.com/html/html_tables.asp

Please watch for IntelliJ IDEA hints (yellow marks on the right, colored strings) - they can help find errors.

File: src/test/java/com/es/phoneshop/Session/HttpSessionCartServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    private ProductDao product;
+
+    @Before
+    public void setup() {
+        cartService = HttpSessionCartService.getInstance();
+        product = ArrayListProductDao.getInstance();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testAddProductToCart() {
```
Comment (SergeiStarikov) [inline src/test/java/com/es/phoneshop/Session/HttpSessionCartServiceTest.java:28]: Please add more tests for edge cases: add product with quantity 1000, etc.

File: src/main/webapp/WEB-INF/pages/productDescription.jsp [hunk 8]
```diff
@@ -5,76 +5,85 @@
...
+                <c:out value="${item}"/>
+                <c:out value="Quantity: ${item.quantity}"/>
+            </p>
+        </c:forEach>
+    </c:if>
+
+    <c:if test="${list!=null}">
+        <c:forEach items="${list}" var="view">
             <table>
                 <td>
```
Comment (SergeiStarikov) [inline src/main/webapp/WEB-INF/pages/productDescription.jsp:77]: Formatting: no <tr> tag

File: src/main/webapp/WEB-INF/pages/productDescription.jsp [hunk 9]
```diff
@@ -5,76 +5,85 @@
...
+            <td> ${products.description}</td>
+            <td>${products.stock}</td>
+            <td class="price">
+                <fmt:formatNumber value="${products.price}" type="currency"
+                                  currencySymbol="${products.currency.symbol}"/>
+            </td>
+        </tr>
+    </table>
+    <form method="post">
+        <label>Quantity</label><input type="text" name="quantity"
```
Comment (SergeiStarikov) [inline src/main/webapp/WEB-INF/pages/productDescription.jsp:44]: HTML formatting
